### PR TITLE
Have scroll position in Quick input stay put in a scenario

### LINKED ISSUE #109969: Quick input reset scroll position
* use latest from master
* f1 > insert snippet
* scroll down to an extension snippet and hide it (press 👁️ icon)
* :bug: the scroll position resets

This is happening when reassigning the items (since the press changed the label) here: https://github.com/microsoft/vscode/blob/92314d61a55f466c125fa9d1f9fe8da633a82423/src/vs/workbench/contrib/snippets/browser/insertSnippet.ts#L213

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
The scenario is... if you set the `.items` on a quickpick with the different items but with the same ids, the quickpick should not scroll. This would fix issues like #109969 where the ItemButtons are being used like a toggle. You would expect the scroll to stay put if you click on an eyeball.

This is the first change to fix the bug internally, then we'll expose this in the API.

Fixes #109969

### CODE REVIEW

File: src/vs/base/parts/quickinput/browser/quickInput.ts [hunk 1]
```diff
@@ -516,9 +516,30 @@ class QuickPick<T extends IQuickPickItem> extends QuickInput implements IQuickPi
...
+			for (const activeItem of this._activeItems) {
+				const item = items.find(i => (i as T).id === activeItem.id);
+				if (item) {
+					newActiveItems.push(item as T);
+				}
+			}
+
+			this._activeItems = newActiveItems;
+			this.activeItemsUpdated = true;
+			scrollTopBefore = (this.ui.list as any).list.scrollTop;
```
Comment (TylerLeonhardt) [inline src/vs/base/parts/quickinput/browser/quickInput.ts:533]: what would be the best way to get this? Add a `scrollTop` prop to `this.ui.list`?

File: src/vs/base/parts/quickinput/browser/quickInput.ts [hunk 2]
```diff
@@ -666,6 +687,44 @@ class QuickPick<T extends IQuickPickItem> extends QuickInput implements IQuickPi
...
+				// verify items have the same id (and that that id is not undefined)
+				if (otherItem.type !== 'separator'
+					&& item.type !== 'separator'
+					&& otherItem.id === item.id
+					&& otherItem.id
+				) {
+					if (ids.has(otherItem.id)) {
+						// duplicate Ids present
+						// TODO: Log to the user properly
+						console.warn('Duplicate IDs present.');
```
Comment (TylerLeonhardt) [inline src/vs/base/parts/quickinput/browser/quickInput.ts:716]: Not sure what the best practice is here

File: src/vs/base/parts/quickinput/browser/quickInput.ts [hunk 3]
```diff
@@ -516,9 +516,30 @@ class QuickPick<T extends IQuickPickItem> extends QuickInput implements IQuickPi
...
+			// Find the new object references of the same id.
+			for (const activeItem of this._activeItems) {
+				const item = items.find(i => (i as T).id === activeItem.id);
+				if (item) {
+					newActiveItems.push(item as T);
+				}
+			}
+
+			this._activeItems = newActiveItems;
+			this.activeItemsUpdated = true;
```
Comment (TylerLeonhardt) [inline src/vs/base/parts/quickinput/browser/quickInput.ts:532]: I might put all of this in an `updatePreservingFocus` function.

**Comment by TylerLeonhardt** [comment]
For tests I can probably reach in to the scrollTop and/or check to make sure the activeItems were updated properly.

**Review by tacosontitan — APPROVED** [review]

**Review by tacosontitan — APPROVED** [review]

**Comment by TylerLeonhardt** [comment]
Closing in favor of #132013

**Pull request closed without merging**